test/ExecutionFramework.cpp

```diff
@@ -170,6 +171,33 @@ void ExecutionFramework::sendMessage(bytes const& _data, bool _isCreation, u256
 		m_transactionSuccessful = (m_gas != m_gasUsed);
 }
 
+void ExecutionFramework::waitForTransaction(std::string const& _txHash) const
+{
+	int polls = 0;
+	while(true)
+	{
+		polls++;
+		auto pendingBlock = m_rpc.eth_getBlockByNumber("pending", false);
+
+		bool isEmpty = pendingBlock["transactions"].empty();
+
+		if (isEmpty && polls < 3000)
+		{
+			if (polls == 200)
+			{
+				cerr << "Note: Already used 200 iterations while waiting for transaction confirmation. Issuing an eth_flush request." << endl;
+				m_rpc.rpcCall("eth_flush");
+			}
+
+			continue;
```

Comment:
I think the control-flow can be simplified here. You can just `return` on success and `BOOST_FAIL` on permanent failure.